### PR TITLE
Fix UT's of util

### PR DESCRIPTION
**Pull request by zBritva**

### CODE REVIEW

File: src/dataLabel/dataLabelUtils.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@
 *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 *  THE SOFTWARE.
 */
+import * as d3 from "d3";
```
Comment (ignatvilesov) [inline src/dataLabel/dataLabelUtils.ts]: Are sure that it's good idea to import everything? Will webpack include whole d3 in such case?

File: src/axis/axis.ts [hunk 2]
```diff
@@ -518,17 +519,20 @@ export function getCategoryThickness(scale: any): number {
...
 
-    let width = scale.rangeBand();
-    let halfInnerPadding = (leftEdges[1] - leftEdges[0] - width) / 2;
+    let leftEdges = [];
+    leftEdges[0] = range[0];
+    leftEdges[1] = range[range.length - 1];
 
-    let range = scale.range();
-    let domain = scale.domain();
+    // let width = scale.bandwidth();
```
Comment (ignatvilesov) [inline src/axis/axis.ts]: Please remove it

File: src/dataLabel/dataLabelUtils.ts [hunk 3]
```diff
@@ -201,23 +200,23 @@ export function getLabelPrecision(precision: number, format: string): number {
...
-    let labels: Update<any> = selectLabels(filteredData, context, false, hasAnimation);
+    let labels: d3.Selection<any, any, any, any> = selectLabels(filteredData, context, false, hasAnimation);
 
     if (!labels) {
         return;
     }
 
     if (hasAnimation) {
-        labels
+        (<any>labels
```
Comment (ignatvilesov) [inline src/dataLabel/dataLabelUtils.ts]: Why we need `any` here?

File: src/legend/interactiveLegend.ts [hunk 4]
```diff
@@ -210,58 +216,60 @@ export class InteractiveLegend implements ILegend {
...
-                }
-            });
-
-        if (this.legendContainerDiv) {
-            this.legendContainerDiv.call(legendZoom);
-        } else {
-            legendTable.call(legendZoom);
-        }
+    private setPanGestureOnLegend(legendTable: d3.Selection<any, any, any, any>): void {
+        throw "Not implmented";
```
Comment (ignatvilesov) [inline src/legend/interactiveLegend.ts]: `Not implemented`.

By the way, what code calls it?

File: test/legendTest.ts [hunk 5]
```diff
@@ -1,1062 +1,1066 @@
...
-                        label: "Washington",
-                        color: "#00ff00",
-                        icon: LegendIcon.Line,
-                        identity: createSelectionIdentity(2),
-                        selected: false
-                    }
-                ];
-            });
+// powerbi.extensibility.utils.chart
+import { } from "./../src/legend/interactiveLegend";
```
Comment (ignatvilesov) [inline test/legendTest.ts]: how it works?

File: src/legend/interactiveLegend.ts (hunk 4 above)
Comment (zBritva) [reply src/legend/interactiveLegend.ts]: It calls in ensureLegendTableCreated method. But it requires converting code related to using d3.ZoomBehavior. I don't know how to properly rewrite setPanGestureOnLegend with d3 v5.

**Pull request merged**